src/mongocxx/options/apm.hpp

```diff
@@ -44,6 +44,9 @@ class MONGOCXX_API apm {
     /// Set the command started monitoring callback. The callback takes a reference to a
     /// command_started_event which will only contain valid data for the duration of the callback.
     ///
+    /// @warning
+    ///   If the callback throws an exception, the behavior is undefined.
```

Comment:
```suggestion
    ///   Callback is expected not to throw. If the callback throws an exception, `std::terminate` is invoked.
```

Suggest updating comments in this file to note throwing an exception is defined to terminate due to `noexcept` on the internal APM helper.

Reply:
For consistency with past/current behavior and to preserve freedom of implementation, I would prefer to leave it publically documented as "undefined behavior" so we do not explicitly guarantee the exact manner in which we will handle (or don't) this scenario for QoI purposes.